group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorPlayback.java

```diff
@@ -38,4 +38,18 @@ void replay(
         short producerEpoch,
         U record
     ) throws RuntimeException;
+
+    /**
+     * Invoke operations when a batch has been successfully loaded.
+     *
+     * @param offset the offset of the last record in the batch plus one.
+     */
+    void updateLastWrittenOffset(Long offset);
```

Comment:
Excuse me, why not use the primitive type `long`? I plan to file a PR to change the type, and please feel free to stop me if I misunderstand anything.